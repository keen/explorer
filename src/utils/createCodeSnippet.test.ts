import {
  createAllCodeSnipped,
  createBodyCode,
  createHeadCode,
} from './createCodeSnippet';

test('creates head code snippet', () => {
  expect(createHeadCode()).toMatchSnapshot();
});

test('creates body code snippet', () => {
  const settings = {
    widget: 'table',
    widgetSettings: {
      title: { content: '@title' },
      subtitle: { content: '@subtitle' },
    },
    chartSettings: {},
    projectId: '@projectId',
    readKey: '@readKey',
    query: {},
  };

  expect(createBodyCode(settings as any)).toMatchSnapshot();
});

test('creates head code snippet', () => {
  const settings = {
    widget: 'table',
    widgetSettings: {
      title: { content: '@title' },
      subtitle: { content: '@subtitle' },
    },
    chartSettings: {},
    projectId: '@projectId',
    readKey: '@readKey',
    query: {},
  };

  expect(createAllCodeSnipped(settings as any)).toMatchSnapshot();
});
