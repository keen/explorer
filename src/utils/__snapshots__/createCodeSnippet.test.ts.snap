// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates body code snippet 1`] = `
"
  <div id=\\"container\\"></div>
  <script type=\\"text/javascript\\">
    const dataviz = new KeenDataviz({
      type: 'table',
      container: '#container',
      widget: {
 \\"title\\": {
  \\"content\\": \\"@title\\"
 },
 \\"subtitle\\": {
  \\"content\\": \\"@subtitle\\"
 }
},
      settings: {},
    });
    
    const client = new KeenAnalysis({
      projectId: '@projectId',
      readKey: '@readKey'
    });
    
    client.query({}).then((res) => dataviz.render(res));
  </script>
"
`;

exports[`creates head code snippet 1`] = `
"
  <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"https://static.keen.io/assets/keen-fonts.css\\" />
  <script crossorigin src=\\"https://cdn.jsdelivr.net/npm/keen-analysis@3\\"></script>
  <script crossorigin src=\\"https://cdn.jsdelivr.net/npm/@keen.io/dataviz@latest/dist/dataviz.min.js\\"></script>
  <style>
    #container {
        height: 300px;
    }
  </style>
"
`;

exports[`creates head code snippet 2`] = `
"
<!doctype html>
<html>
  <head>
    <meta charset=\\"utf-8\\">
    <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"https://static.keen.io/assets/keen-fonts.css\\" />
    <script crossorigin src=\\"https://cdn.jsdelivr.net/npm/keen-analysis@3\\"></script>
    <script crossorigin src=\\"https://cdn.jsdelivr.net/npm/@keen.io/dataviz@latest/dist/dataviz.min.js\\"></script>
    <style>
      #container {
        height: 300px;
      }
    </style>
  </head>
  <body>
  <div id=\\"container\\"></div>
  <script type=\\"text/javascript\\">
    const dataviz = new KeenDataviz({
      type: 'table',
      container: '#container',
      widget: {
 \\"title\\": {
  \\"content\\": \\"@title\\"
 },
 \\"subtitle\\": {
  \\"content\\": \\"@subtitle\\"
 }
},
      settings: {},
    });
    
    const client = new KeenAnalysis({
      projectId: '@projectId',
      readKey: '@readKey'
    });
    
    client.query({}).then((res) => dataviz.render(res));
    
  </script>
  </body>
</html>
"
`;
